### PR TITLE
Update deploy and auth documentation to use menuSuggestionFlow

### PR DESCRIPTION
**Pull request by MaesterChestnut**

### CODE REVIEW

File: docs/dotprompt.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ input:
 ---
 
 {{role "system"}}
-You are a helpful AI assistant that really loves to talk about puppies. Try to work puppies
+You are a helpful AI assistant that really loves to talk about food. Try to work food
```
Comment (tagboola) [inline docs/dotprompt.md:202]: "Try to work food" 🤔

**Review by tagboola — APPROVED** [review]

**Review by tagboola — APPROVED** [review]

File: docs/dotprompt.md [hunk 2]
```diff
@@ -42,12 +42,12 @@ import { prompt } from '@genkit-ai/dotprompt';
 Then, load the prompt using `prompt('file_name')`:
 
 ```ts
-const greetingPrompt = await prompt('greeting');
+const menuPrompt = await prompt('menu');
```
Comment (MichaelDoyle) [inline docs/dotprompt.md:45]: We're calling this `menuPrompt` here, but it looks like the prompt above is still providing a greeting in the style of...

File: docs/firebase.md [hunk 3]
```diff
@@ -277,8 +280,8 @@ app:
             .querySelector('#signinBtn')
             .addEventListener('click', signIn);
           document
-            .querySelector('#tellJoke')
-            .addEventListener('click', generateJoke);
+            .querySelector('#suggestMenuItem')
+            .addEventListener('click', generateMenuItem);
```
Comment (MichaelDoyle) [inline docs/firebase.md:285]: I believe there is a webapp that goes along with all of this, hence why all of the dom related methods here. Are we updating that as well?

**Comment by MichaelDoyle** [comment]
I suggest we split this into 2 PRs.

One that ensures that the "deployment" related docs work as intended with `genkit init` (these are firebase.md, cloud-run.md and deploy-node.md), and another that does more of the snippet standardization work.

File: docs/deploy-node.md [hunk 4]
```diff
@@ -104,7 +104,7 @@ sample flow.
     Then, in another window:
 
     ```posix-terminal
-    curl -X POST "http://127.0.0.1:3400/jokeFlow?stream=true" -H "Content-Type: application/json"  -d '{"data": "banana"}'
+    curl -X POST "http://127.0.0.1:3400/menuSuggestionFlow?stream=true" -H "Content-Type: application/json"  -d '{"data": "banana"}'
```
Comment (MichaelDoyle) [inline docs/deploy-node.md:107]: Should this still take "banana" as input?

File: docs/deploy-node.md [hunk 5]
```diff
@@ -81,7 +81,7 @@ sample flow.
     ```posix-terminal
     npm run build
 
-    genkit flow:run jokeFlow "\"banana\"" -s
+    genkit flow:run menuSuggestionFlow "\"banana\"" -s
```
Comment (MichaelDoyle) [inline docs/deploy-node.md:84]: Should this still take "banana" as input?

File: docs/auth.md [hunk 6]
```diff
@@ -140,7 +140,7 @@ above. When running this flow during development, you would pass the user object
 in the same way:
 
 ```posix-terminal
-genkit flow:run jokeFlow '"Banana"' --auth '{"admin": true}'
+genkit flow:run menuSuggestionFlow '"Banana"' --auth '{"admin": true}'
```
Comment (MichaelDoyle) [inline docs/auth.md:143]: Should this still take "banana" as input?

File: docs/cloud-run.md [hunk 7]
```diff
@@ -122,7 +122,7 @@ flow.
 
     1.  In the developer UI (http://localhost:4000/), run the flow:
 
-        1.  Click **jokeFlow**.
```
Comment (MichaelDoyle) [inline docs/cloud-run.md:125]: There are still some references below that need to be updated.

L129-131 - subject matter doesn't work as well w/ menuSuggestionFlow

L162-166 curls jokeFlow w/ banana as input.

File: docs/auth.md (hunk 6 above)
Comment (MichaelDoyle) [reply docs/auth.md:143]: (if we're still going w/ menuSuggestionFlow)

File: docs/auth.md [hunk 8]
```diff
@@ -190,8 +190,8 @@ the following configuration options to your `onFlow()`:
 ```ts
 import { onFlow } from '@genkit-ai/firebase/functions';
 
-export const jokeFlow = onFlow({
-    name: 'jokeFlow',
+export const menuSuggestionFlow = onFlow({
+    name: 'menuSuggestionFlow',
```
Comment (MichaelDoyle) [inline docs/auth.md:194]: Reference to `(joke)` a few more lines below as an input param under authPolicy.

**Review by kevinthecheung — APPROVED** [review]

**Review by MichaelDoyle — APPROVED** [review]
image

**Pull request merged**